### PR TITLE
JSUI-3238 2nd PR Smart snippet without iframe

### PR DESCRIPTION
**Pull request by erocheleau**
Fixing the jsdoc comments ... sorry

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/misc/AttachShadowPolyfill.ts [hunk 1]
```diff
@@ -4,24 +4,34 @@ import 'styling/_AttachShadowPolyfill';
 export interface IShadowOptions {
   title: string;
   onSizeChanged?: Function;
+  useIFrame?: Boolean;
```
Comment (samisayegh) [inline src/misc/AttachShadowPolyfill.ts:7]: Should the `f` be lowercase?

File: src/misc/AttachShadowPolyfill.ts [hunk 2]
```diff
@@ -4,24 +4,34 @@ import 'styling/_AttachShadowPolyfill';
...
+  let contentBody: HTMLElement;
+  if (options.useIFrame) {
+    const iframe = $$('iframe', { className: 'coveo-shadow-iframe', scrolling: 'no', title: options.title }).el as HTMLIFrameElement;
+    const onLoad = new Promise(resolve => iframe.addEventListener('load', () => resolve()));
+    element.appendChild(iframe);
+    await onLoad;
+    contentBody = iframe.contentDocument.body as HTMLBodyElement;
+    autoUpdateContainer = iframe;
+  } else {
+    autoUpdateContainer = $$('div', { className: 'coveo-shadow-iframe', scrolling: 'no', title: options.title }).el as HTMLElement;
```
Comment (samisayegh) [inline src/misc/AttachShadowPolyfill.ts:21]: Could this line be reused? Seems like a duplicate of line 14?

**Review by samisayegh — APPROVED** [review]

File: src/misc/AttachShadowPolyfill.ts (hunk 1 above)
Comment (erocheleau) [reply src/misc/AttachShadowPolyfill.ts:7]: I always thought it was an iFrame ?

`use-i-frame` or `use-iframe` in the html option?
I feel like people might mistype the 2nd one and try to do `use-frame` instead? What do you think?

Comment (samisayegh) [reply src/misc/AttachShadowPolyfill.ts:7]: honestly, the second looks really natural to me, _but_ I see that the typescript is `HTMLIFrameElement`.
The current casing is consistent with the type, so leaving it as-is makes sense 👍

**Review by btaillon — APPROVED** [review]

**Review by lvu285 — APPROVED** [review]

**Review by olamothe — APPROVED** [review]

**Pull request merged**